helix-term/src/commands.rs

```diff
@@ -2202,10 +2211,16 @@ fn symbol_picker(cx: &mut Context) {
                         if let Some(range) =
                             lsp_range_to_range(doc.text(), symbol.location.range, offset_encoding)
                         {
-                            doc.set_selection(view.id, Selection::single(range.to(), range.from()));
+                            doc.set_selection(view.id, Selection::from(range));
                             align_view(doc, view, Align::Center);
                         }
                     },
+                    move |editor, symbol| {
+                        let view = editor.tree.get(editor.tree.focus);
+                        let doc = &editor.documents[view.doc];
+                        let line = Some(symbol.location.range.start.line as usize);
```

Comment:
I don't think the symbol is guaranteed to be in the current file, you should use `symbol.location.uri` to look up the doc (you can use `uri.to_file_path` to compare).

Reply:
Actually, you can probably just do what you did in goto_impl

Reply:
Done; also I noticed a slight issue with the gotos: `location.range.start.line` returns the start of the whole item. So instead of the line with the symbol itself, usually the documentation's first line is returned. I remember some attribute for getting the position of the symbol itself, but it was only for nested document symbols ? Not sure, will have to look it up (also I think it can be tackled in a separate PR).

Reply:
Yeah, there seems to be two types, `DocumentSymbol` and `SymbolInformation`, the latter one being used here. I think you could just highlight the entire line range from start to end.